src/Stratis.Bitcoin.Features.SmartContracts/Rules/AllowedScriptTypeRule.cs

```diff
@@ -60,10 +62,19 @@ private void CheckOutput(TxOut output)
             if (PayToPubkeyHashTemplate.Instance.CheckScriptPubKey(output.ScriptPubKey))
                 return;
 
+            // For cross-chain transfers
+            if (PayToScriptHashTemplate.Instance.CheckScriptPubKey(output.ScriptPubKey))
+                return;
+
             // For cross-chain transfers
             if (PayToMultiSigTemplate.Instance.CheckScriptPubKey(output.ScriptPubKey))
                 return;
 
+            // For cross-chain transfers
+            IEnumerable<Op> ops = output.ScriptPubKey.ToOps();
```

Comment:
I think there is a template for opreturn `TxNullDataTemplate`

Reply:
Ah the problem here is that MAX_OP_RETURN_RELAY is different for Stratis so we may need to fix it a bit.

Reply:
Hmm, have put this in. In terms of just checking the script template type, I believe this will work.